### ISSUE TITLE
Clean up delphi_utils

### ISSUE THREAD

**Issue by melange396** [cmu-delphi/covidcast-indicators#1973]
`delphi_utils` is a big hodge-podge of a package with a lot of dependencies, thus it has a relatively large net footprint and takes a while to install.  Lets see if we can lessen that.

Move [packages in setup.py](https://github.com/cmu-delphi/covidcast-indicators/blob/a6ea00355d79a47cc09db8090f6e5121c718de0f/_delphi_utils_python/setup.py#L8-L26) that are unnecessary for a downstream user but still used in test/lint/ci, like: `darker`, `freezegun`, `mock`, `moto`, `pydocstyle`, `pylint`, `pytest-cov`, `pytest`, `requests-mock`.  They can be put into something like a "`requirements.test.txt`" file that gets installed via [the CI](https://github.com/cmu-delphi/covidcast-indicators/blob/a6ea00355d79a47cc09db8090f6e5121c718de0f/.github/workflows/python-ci.yml#L53-L70) and/or [Makefile](https://github.com/cmu-delphi/covidcast-indicators/blob/a6ea00355d79a47cc09db8090f6e5121c718de0f/_delphi_utils_python/Makefile#L11-L25).  

Remove from `setup.py` any unused packages, as `gitpython` appears to be.

The `covidcast` package has its own [wacky imports](https://github.com/cmu-delphi/covidcast/blob/main/Python-packages/covidcast-py/setup.py#L23-L33), which includes a bunch of imaging/graphics and geospatial libraries.  It is only used in the `delphi_utils.validator` module for pulling data from the API and can be replaced with the lighter-weight [`delphi_epidata` client](https://pypi.org/project/delphi-epidata/) instead.  Doing that is not quite as easy as moving requirements entries around, and it is covered in in #1972.

**Comment by melange396** [comment]
We could even put some of the esoteric usage and/or larger installation dependencies behind [package "extras"](https://packaging.python.org/en/latest/tutorials/installing-packages/#installing-extras).

**Comment by dshemetov** [comment]
If we combine this with migrating over to `pyproject.toml`, we could use the [`optional-dependencies`](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#dependencies-optional-dependencies) interface to separate out the `dev` dependencies. See [here](https://github.com/cmu-delphi/delphi-epidata/pull/1470) for an example of migrating over to `pyproject.toml`.

**Comment by melange396** [comment]
migrating to using `pyproject.toml` helps unlock this issue: https://github.com/cmu-delphi/covidcast-indicators/issues/1839

**Comment by dshemetov** [comment]
Gonna reopen this issue. Also, gonna edit your OP @melange396 so the subtasks are clearer because #1984 only addressed one of the subtasks

**Comment by dshemetov** [comment]
Thinking about the unused package subtask, I found [`deptry`](https://github.com/fpgmaas/deptry) and ran it on _delphi_utils and it reported this:

```
(env) ➜  _delphi_utils_python git:(1972-replace-covidcast) ✗ deptry .
Scanning 31 files...

data_proc/geomap/geo_data_proc.py:13:8: DEP003 'requests' imported but it is a transitive dependency
data_proc/geomap/source-file-sanity-check.ipynb:23:8: DEP003 'matplotlib' imported but it is a transitive dependency
delphi_utils/flash_eval/eval_day.py:6:1: DEP003 'scipy' imported but it is a transitive dependency
delphi_utils/validator/datafetcher.py:13:8: DEP003 'requests' imported but it is a transitive dependency
requirements.txt: DEP002 'scs' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'darker' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'freezegun' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'mock' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'moto' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'pydocstyle' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'pylint' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'pytest-cov' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'pytest' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'requests-mock' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'xlrd' defined as a dependency but not used in the codebase
Found 16 dependency issues.

For more information, see the documentation: https://deptry.com/
```

These will be handled by #2014, as they get moved to dev.

```
requirements.txt: DEP002 'darker' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'freezegun' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'mock' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'moto' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'pydocstyle' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'pylint' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'pytest-cov' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'pytest' defined as a dependency but not used in the codebase
requirements.txt: DEP002 'requests-mock' defined as a dependency but not used in the codebase
```

- `xlrd` is needed by Pandas (maybe we should write [`pandas[excel]`](https://pandas.pydata.org/docs/getting_started/install.html#excel-files) so this dependence is explicit?). 
- `scs` is explicitly version pinned for `cvxpy` because of upstream issues
- Seems like a good idea to explicitly list `requests` as a dependecy
- flash we can probably ignore until needed and geomap scripts we can ignore until needed

---

TLDR: 
- [ ] replace `pandas` with `pandas[excel]` and add `requests` explicitly to the dependecy list
